controlBoardRemapper: align the mandatory interfaces to controlBoard_nws_yarp

Nicogene opened the pull request:
As per `controlBoard_nws_yarp` it is mandatory only that at list is implemented `iAxisInfo` **OR** `iPositionControl`.

randaz81 reviewed (changes requested):
It seems to me that the not-mandatory interfaces are not adequately protected elsewhere.
For example, if `vel` interface is now optional, it must have a guard `if (ivel!=nullptr)` in ControlBoardRemapper.cpp to prevent a segfault if a client is calling it

randaz81 commented:
I'd suggest also to add a new test, which attaches to `fakeMotionControlMicro` instead of `fakeMotionControl`.
This because fakeMotionControl actually implements all interfaces, so the current test cannot catch any issue if the remapper is used with a device that does not implement all interfaces.

Nicogene commented:
@randaz81 @traversaro I added all the safeguards for those interfaces that are not mandatory anymore.
Moreover, I added a test on the `fakeMotionControlMicro` in order to check these new changes, finally I successfully tested the `controlBoardRemapper` in simulation.
